### PR TITLE
support passing user-info to AOAI for Defender for cloud

### PR DESCRIPTION
**Pull request by slreznit**
### Motivation and Context

Defender for AI is part of Defender for Cloud and to provide advanced threat protection to customers we need gen-ai apps to be instrumented as this PR suggests

This PR is part1 - Add functionality and make it opt-in to start tests and do some demos.
I plan to add a check whether Defender for AI is enabled in another PR

### Contribution Checklist

- [ ] I have built and tested the code locally and in a deployed app
- [ ] For frontend changes, I have pulled the latest code from main, built the frontend, and committed all static files.
- [ ] This is a change for all users of this app. No code or asset is specific to my use case or my organization.
- [ ] I didn't break any existing functionality :smile:

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -733,6 +733,18 @@ def prepare_model_args(request_body):
         if message:
             messages.append({"role": message["role"], "content": message["content"]})
 
+    authenticated_user_details = get_authenticated_user_details(request_headers)
```
Comment (sarah-widder) [inline app.py:736]: should this be conditionally enabled? or only if the developer has opted in?

File: app.py [hunk 2]
```diff
@@ -723,7 +723,7 @@ def get_configured_data_source():
     return data_source
 
 
-def prepare_model_args(request_body):
+async def prepare_model_args(request_body, request_headers):
```
[the remark this answers is not in the document]
Comment (slreznit) [reply app.py:727]: fixed

File: app.py (hunk 1 above)
Comment (slreznit) [reply app.py]: Added MS_DEFENDER_ENABLED env var to enable this as first step

**Review by abhahn — COMMENTED** [review]
Just wondering -- will any of these values being null or empty string cause errors from the service?  If there are any known error conditions we should be aware of here it might be helpful to add some info about them to the README.

**Comment by slreznit** [comment]
> Just wondering -- will any of these values being null or empty string cause errors from the service? If there are any known error conditions we should be aware of here it might be helpful to add some info about them to the README.

The user field is a string and will not fail the API call if inner json has null or empty values.

File: app.py [hunk 3]
```diff
@@ -733,6 +734,20 @@ def prepare_model_args(request_body):
         if message:
             messages.append({"role": message["role"], "content": message["content"]})
 
+    user_json = None
+    if (MS_DEFENDER_ENABLED):
+        authenticated_user_details = get_authenticated_user_details(request_headers)
+        tenantId = get_tenantid(authenticated_user_details.get("client_principal_b64"))
+        conversation_id = request_body.get("conversation_id", None)        
+        user_args = {
```
Comment (abhahn) [inline app.py:742]: `user_args` is what my previous comment was referring to.  I think I just forgot to click here to add that comment.  :)

Comment (slreznit) [reply app.py:742]: Right! 
I did test it with None values without any issue

**Review by abhahn — APPROVED** [review]

**Pull request merged**